### PR TITLE
feat(TDP-2933): about modal

### PR DESCRIPTION
**Pull request by amaalej**
**Link to the JIRA issue**
e.g. https://jira.talendforge.org/browse/TDP-2933

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
In react route there is no footer.

**(Optional) What is the new behavior?**
Instead of a footer, there is modal showing the version, copyrights and *builds details*

**(Optional) Other information:**
Linked to EE PR: https://github.com/Talend/data-prep-ee/pull/95

### CODE REVIEW

File: dataprep-webapp/src/app/components/about/about.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  ~  You should have received a copy of the agreement
+  ~  along with this program; if not, write to Talend SA
+  ~  9 rue Pages 92150 Suresnes, France
+  ~
+  ~ ============================================================================
+  -->
+
+<talend-modal
+        id="about-modal"
+        fullscreen="false"
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about.html]: Remove that

File: dataprep-webapp/src/app/components/about/about.html [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        close-button="true"
+        ng-if="$ctrl.state.home.showAbout"
+        state="$ctrl.state.home.showAbout"
+        disableCloseOnBackgroundClick="false"
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about.html]: Remove that

File: dataprep-webapp/src/app/components/about/about.html [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        disable-enter="true">
+
+    <div class="modal-header">
+        {{'ABOUT_HEADER' | translate}}
+    </div>
+
+    <div class="modal-body">
+        <span id="about-dataprep-logo" data-icon="c"></span>
+        <span class="version-copyrights">
+            <span id="version">{{"VERSION" | translate}}: {{$ctrl.version}}</span>
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about.html]: This should be centered

File: dataprep-webapp/src/app/components/about/about.html [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            <tbody>
+                <tr ng-repeat="buildDetail in $ctrl.buildDetails track by $index">
+                    <td>{{buildDetail.serviceName}}</td>
+                    <td>{{buildDetail.buildId}}</td>
+                    <td>{{buildDetail.versionId}}</td>
+                </tr>
+            </tbody>
+        </table>
+
+        <button class="btn btn-primary" translate-once="MORE" ng-if="!$ctrl.showBuildDetails" ng-click="$ctrl.getBuildDetails()"></button>
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about.html]: Get the build infos only once on component creation. Just toggle on click.

File: dataprep-webapp/src/app/components/about/about-module.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import ngTranslate from 'angular-translate';
+
+import AboutComponent from './about-component.js';
+import SERVICES_UTILS_MODULE from '../../services/utils/utils-module';
+import SERVICES_ABOUT_MODULE from '../../services/about/about-module';
+import SERVICES_STATE_MODULE from '../../services/state/state-module';
+
+const MODULE_NAME = 'data-prep.about';
+
+angular.module(MODULE_NAME, [ngTranslate, SERVICES_UTILS_MODULE, SERVICES_ABOUT_MODULE, SERVICES_STATE_MODULE])
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about-module.js]: Indent please

File: dataprep-webapp/src/app/components/about/about-module.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+ ============================================================================*/
+
+import angular from 'angular';
+import ngTranslate from 'angular-translate';
+
+import AboutComponent from './about-component.js';
+import SERVICES_UTILS_MODULE from '../../services/utils/utils-module';
+import SERVICES_ABOUT_MODULE from '../../services/about/about-module';
+import SERVICES_STATE_MODULE from '../../services/state/state-module';
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about-module.js]: Sort please

File: dataprep-webapp/src/app/components/about/about-component.spec.js [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+	beforeEach(angular.mock.module('data-prep.about', ($provide) => {
+		stateMock = {
+			home: {
+				showAbout: true,
+			},
+		};
+		$provide.constant('state', stateMock);
+	}));
+
+	beforeEach(angular.mock.module('pascalprecht.translate', function ($translateProvider) {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/about-component.spec.js]: Prefer arrow function

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+			top: 1em;
+			right: 1em;
+
+			&:before, &:after {
+				background-color: $black;
+			}
+		}
+
+		.modal-content {
+			border: 0;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: Is this really necessary ?

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+		}
+
+		.modal-content {
+			padding: 0;
+			width: 20%;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: On 1300px resolution, this looks bad

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+			margin: 8rem auto;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: Why is it different from all the other modals ?

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+			.modal-header {
+				text-align: center;
+				color: $black;
+				opacity: .8;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: Why ?

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+				border-radius: 5px;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: ??

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+				letter-spacing: .5px;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: Why ?

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+				font-weight: 500;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: Why ?

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+				background-color: $gray;
+			}
+
+			.modal-body {
+				@include display(flex);
+				@include flex-direction(column);
+				@include align-items(center);
+				@include justify-content(space-between);
+				height: 100%;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]:
```
@include display(flex);
@include flex-direction(column);
@include align-items(center);
@include justify-content(space-between);
height: 100%;
```

A lot of unnecessary things here, remove all that and replace by `text-align: center`

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+
+				#about-dataprep-logo {
+					font-size: 2em;
+					color: $blue;
+				}
+
+				.version-copyrights {
+					@include display(flex);
+					@include flex-direction(column);
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]:
```
@include display(flex);
@include flex-direction(column);
```

Remove that and replace the spans by divs to

File: dataprep-webapp/src/app/components/about/_about.scss [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+					margin: 15px 0;
+				}
+
+				table {
+					margin: 10px;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/about/_about.scss]: `margin: auto` to center it

File: dataprep-webapp/src/app/services/about/rest/about-rest-service.js [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+
+ ============================================================================*/
+
+/**
+ * @ngdoc service
+ * @name data-prep.services.about.service:AboutRestService
+ * @description About service. This service provides the list of the build ids of each backend service
+ * @requires data-prep.services.utils.service:RestURLs
+ */
+export default class AboutRestService {
```
Comment (amaalej) [inline dataprep-webapp/src/app/services/about/rest/about-rest-service.js]: What do you mean by *expose a method that get the versions and set them in state * the versions should be set in the state?

**Comment by amaalej** [comment]
@jsomsanith done.

**Pull request merged**